torch/csrc/autograd/utils/python_variables.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include <ATen/ATen.h>
+#include "torch/csrc/autograd/python_variable.h"
+
+namespace torch { namespace autograd { namespace utils {
+
+inline at::Tensor set_requires_grad(at::Tensor self, bool requires_grad) {
```

Comment:
Why don't we put this into `Variable::set_requires_grad`?

Reply:
I didn't do this because I plan on making this a python warning in a future commit if people don't object (I don't like errors on cases meaning "this is probably not what you meant" during prototyping).

Reply:
to clarify: a warning that the request is being ignored, i.e. requires_grad will be false.

Reply:
I don't think we ever want to raise warnings instead of errors. As a user I'd like to know if I'm doing something wrong. Otherwise you need to scroll through logs of your program and look for warnings, which is especially annoying if it's producing a lot of output.

Reply:
Existing code likely didn't do what the users intended, so I'm pretty happy it will be broken. The reason why we even had the previous PR is because someone thought that not being able to differentiate w.r.t. integers is a PyTorch bug.

Reply:
Sure, although I'd argue the error is trying to backprop through an integer tensor with requires grad, not just creating one.

Reply:
It's hard to provide a meaningful error message in that case. You'll only be able to tell someone that they messed up *somewhere*, while raising an error immediately will point them right to the location where they made the mistake.